wp-includes/widgets/class-wp-widget-visual-text.php

```diff
@@ -0,0 +1,160 @@
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param array $args     Display arguments including 'before_title', 'after_title',
+	 *                        'before_widget', and 'after_widget'.
+	 * @param array $instance Settings for the current Text widget instance.
+	 */
+	public function widget( $args, $instance ) {
+
+		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		$widget_text = ! empty( $instance['text'] ) ? $instance['text'] : '';
+
+		/**
+		 * Filters the content of the Text widget.
+		 *
+		 * @since 2.3.0
+		 * @since 4.4.0 Added the `$this` parameter.
+		 *
+		 * @param string         $widget_text The widget content.
+		 * @param array          $instance    Array of settings for the current widget.
+		 * @param WP_Widget_Text $this        Current Text widget instance.
+		 */
+		$text = apply_filters( 'widget_text', $widget_text, $instance, $this );
+
+		// Apply visual text filters for widgets that have been created or updated since 4.8, wherein the 'filter' prop is removed.
+		if ( ! isset( $instance['filter'] ) ) {
+
+			/**
+			 * Filters the content of the Text widget to apply changes expected from the visual (TinyMCE) editor.
+			 *
+			 * By default a subset of the_content filters are applied, including wpautop and wptexturize.
+			 *
+			 * @since 4.8.0
+			 *
+			 * @param string         $widget_text The widget content.
+			 * @param array          $instance    Array of settings for the current widget.
+			 * @param WP_Widget_Text $this        Current Text widget instance.
+			 */
+			$text = apply_filters( 'widget_text_content', $widget_text, $instance, $this );
+
+		} elseif ( true === $instance['filter'] ) {
+			$text = wpautop( $text ); // Back-compat for instances prior to 4.8.
+		}
+
+		echo $args['before_widget'];
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . $title . $args['after_title'];
+		}
+
+		?>
+			<div class="textwidget"><?php echo $text; ?></div>
+		<?php
+		echo $args['after_widget'];
+	}
+
+	/**
+	 * Handles updating settings for the current Text widget instance.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param array $new_instance New settings for this instance as input by the user via
+	 *                            WP_Widget::form().
+	 * @param array $old_instance Old settings for this instance.
+	 * @return array Settings to save or bool false to cancel saving.
+	 */
+	public function update( $new_instance, $old_instance ) {
+		$instance = $old_instance;
+		$instance['title'] = sanitize_text_field( $new_instance['title'] );
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			$instance['text'] = $new_instance['text'];
+		} else {
+			$instance['text'] = wp_kses_post( $new_instance['text'] );
+		}
+
+		// Eliminate filter from here on.
+		unset( $instance['filter'] );
```

Comment:
Maybe this is not a good idea, for example it could cause problems if a WP install downgrades from 4.8 to 4.7. Instead of removing `filter` maybe we should instead force `filter` to always be `true` and then to _add_ a new property, like `visual` which also gets set to `true`. Then when calling `widget` to render an instance, we can check for the presence of `visual` instead of `! isset( $instance['filter'] )`. Then the `filter` instance property would just persist as a back-compat property during 4.8 and could potentially be removed in the future.